source/common/upstream/load_balancer_impl.cc

```diff
@@ -517,20 +522,37 @@ bool ZoneAwareLoadBalancerBase::earlyExitNonLocalityRouting() {
   return false;
 }
 
-bool LoadBalancerContextBase::validateOverrideHostStatus(Host::Health health,
-                                                         OverrideHostStatus status) {
-  switch (health) {
-  case Host::Health::Unhealthy:
-    return status & UnhealthyStatus;
-  case Host::Health::Degraded:
-    return status & DegradedStatus;
-  case Host::Health::Healthy:
-    return status & HealthyStatus;
+uint32_t LoadBalancerContextBase::createOverrideHostStatus(
+    const envoy::config::cluster::v3::Cluster::CommonLbConfig& common_config) {
+  if (!common_config.has_override_host_status()) {
+    return DefaultOverrideHostStatus;
+  }
+
+  uint32_t override_host_status = 0;
+
+  for (auto single_status : common_config.override_host_status().statuses()) {
+    switch (static_cast<envoy::config::core::v3::HealthStatus>(single_status)) {
+    case envoy::config::core::v3::HealthStatus::UNKNOWN:
+    case envoy::config::core::v3::HealthStatus::HEALTHY:
+      override_host_status |= singleHealthStatusToUint(Host::Health::Healthy);
+      break;
+    case envoy::config::core::v3::HealthStatus::UNHEALTHY:
+    case envoy::config::core::v3::HealthStatus::DRAINING:
+    case envoy::config::core::v3::HealthStatus::TIMEOUT:
+      override_host_status |= singleHealthStatusToUint(Host::Health::Unhealthy);
+      break;
+    case envoy::config::core::v3::HealthStatus::DEGRADED:
+      override_host_status |= singleHealthStatusToUint(Host::Health::Degraded);
+      break;
+    default:
+      break;
```

Comment:
I think this may hide a case where the server sends a config with some other health status.
Consider adding `[(validate.rules).enum.defined_only = true]` to the `statuses`, and maybe a emit a log here.

Reply:
get it.